### PR TITLE
Fix some issues with parcelable code gen with enums and flags

### PR DESCRIPTION
**Pull request by 4brunu**
The Flag type was added, but the support for parcelable for it was missing, so this PR add's it.
This also fixes https://github.com/dropbox/djinni/issues/431 and improves the test coverage for parcelable.

### CODE REVIEW

**Comment by 4brunu** [comment]
@artwyman and @xianwen can you please take a look at this?
Thanks

File: example/example.djinni [hunk 1]
```diff
@@ -1,6 +1,6 @@
 item_list = record {
     items: list<string>;
-}
+} deriving(parcelable)
```
Comment (artwyman) [inline example/example.djinni]: Why is this needed in the example?  Seems unnecessary, and the example should stay simple and understandable.
Given the generated code isn't updated either, I suspect maybe this was a test case you didn't intend to push.

File: example/run_djinni.sh [hunk 2]
```diff
@@ -55,6 +55,7 @@ fi
     --java-package $java_package \
     --java-class-access-modifier "package" \
     --java-generate-interfaces true \
+    --java-implement-android-os-parcelable true \
```
Comment (artwyman) [inline example/run_djinni.sh]: As above, I think this feature shouldn't be included in the basic example.

Comment (4brunu) [reply example/run_djinni.sh]: I included this here, because this way, if someone is developing a new feature in djinni and that feature breaks parcelable, it will notice faster, but if you want, I can remove it.

Comment (artwyman) [reply example/run_djinni.sh]: A good goal, though I think the test-suite covers tat, without the example having to be involved.  As-is, the example doesn't build for non-Android Java, which is one of the included build flavors.

**Comment by prsolucoes** [comment]
It will be merged too?

**Comment by 4brunu** [comment]
@artwyman All your suggestions are done 👍
